components/encoder/encoder_test.go

```diff
@@ -0,0 +1,295 @@
+	test.That(t, actualEncoder1.reconfCount, test.ShouldEqual, 1)
+
+	test.That(t, actualEncoder1.posCount, test.ShouldEqual, 0)
+	test.That(t, actualEncoder2.posCount, test.ShouldEqual, 0)
+	result, _, err := reconfEncoder1.(encoder.Encoder).GetPosition(context.Background(), pb.PositionType_POSITION_TYPE_UNSPECIFIED.Enum(), nil)
+	test.That(t, err, test.ShouldBeNil)
+	test.That(t, result, test.ShouldResemble, position)
+	test.That(t, actualEncoder1.posCount, test.ShouldEqual, 0)
+	test.That(t, actualEncoder2.posCount, test.ShouldEqual, 1)
+
+	err = reconfEncoder1.Reconfigure(context.Background(), nil)
+	test.That(t, err, test.ShouldNotBeNil)
+	test.That(t, err, test.ShouldBeError, rutils.NewUnexpectedTypeError(reconfEncoder1, nil))
+
+	actualEncoder3 := &mock{Name: failEncoderName}
+	reconfEncoder3, err := encoder.WrapWithReconfigurable(actualEncoder3, resource.Name{})
+	test.That(t, err, test.ShouldBeNil)
+	test.That(t, reconfEncoder3, test.ShouldNotBeNil)
+}
+
+func TestResetPosition(t *testing.T) {
+	actualEncoder1 := &mock{Name: testEncoderName}
+	reconfEncoder1, err := encoder.WrapWithReconfigurable(actualEncoder1, resource.Name{})
+	test.That(t, err, test.ShouldBeNil)
+
+	test.That(t, actualEncoder1.zeroCount, test.ShouldEqual, 0)
+	err = reconfEncoder1.(encoder.Encoder).ResetPosition(context.Background(), nil)
+	test.That(t, err, test.ShouldBeNil)
+	test.That(t, actualEncoder1.zeroCount, test.ShouldEqual, 1)
+}
+
+func TestGetPosition(t *testing.T) {
+	actualEncoder1 := &mock{Name: testEncoderName}
+	reconfEncoder1, err := encoder.WrapWithReconfigurable(actualEncoder1, resource.Name{})
+	test.That(t, err, test.ShouldBeNil)
+
+	test.That(t, actualEncoder1.posCount, test.ShouldEqual, 0)
+	pos1, positionType, err := reconfEncoder1.(encoder.Encoder).GetPosition(context.Background(), nil, nil)
+	test.That(t, err, test.ShouldBeNil)
+	test.That(t, pos1, test.ShouldResemble, position)
+	test.That(t, actualEncoder1.posCount, test.ShouldEqual, 1)
+	test.That(t, positionType, test.ShouldEqual, pb.PositionType_POSITION_TYPE_UNSPECIFIED)
+
+	props, err := reconfEncoder1.(encoder.Encoder).GetProperties(context.Background(), nil)
+	test.That(t, err, test.ShouldBeNil)
+	test.That(t, props[encoder.TicksCountSupported], test.ShouldBeTrue)
+	test.That(t, props[encoder.AngleDegreesSupported], test.ShouldBeTrue)
+}
+
+func TestClose(t *testing.T) {
+	actualEncoder1 := &mock{Name: testEncoderName}
+	reconfEncoder1, err := encoder.WrapWithReconfigurable(actualEncoder1, resource.Name{})
+	test.That(t, err, test.ShouldBeNil)
+
+	test.That(t, actualEncoder1.reconfCount, test.ShouldEqual, 0)
+	test.That(t, utils.TryClose(context.Background(), reconfEncoder1), test.ShouldBeNil)
+	test.That(t, actualEncoder1.reconfCount, test.ShouldEqual, 1)
+}
+
+var (
+	position = 5.5
+	features = map[encoder.Feature]bool{
+		encoder.TicksCountSupported:   true,
+		encoder.AngleDegreesSupported: true,
+	}
+)
+
+type mock struct {
```

Comment:
[if-minor] I think we can use inject.Encoder for this purpose; `setupInjectRobot` can return a state object with these fields that gets mutated by the injected methods via closure. Then you can check the values on the state in the tests. Helps with codebase consistency if everything is using inject, but if it turns out to be too much trouble don't force it

Reply:
leaving this as is for now